### PR TITLE
DEV-20247 Fix return value for OGR::Layer::Extensions#point_values wh…

### PR DESCRIPTION
**Pull request by tindron**
…en no features are found

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -6,19 +6,24 @@ on:
 jobs:
   test:
     name: Test
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ 2.6, 2.7, '3.0', 3.1 ]
```
Comment (kurtronshausen) [inline .github/workflows/continuous-integration.yml:12]: why does 3.0 need quotes?

File: lib/ogr/layer/extensions.rb [hunk 2]
```diff
@@ -113,7 +113,7 @@ def geometry_from_extent
       #   from.
       # rubocop:todo Metrics/PerceivedComplexity
       def point_values(with_attributes = {}) # rubocop:todo Metrics/CyclomaticComplexity
-        return [[]] if feature_count.zero?
+        return [] if feature_count.zero?
```
Comment (kurtronshausen) [inline lib/ogr/layer/extensions.rb:116]: is this the actual bug fix?

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (tindron) [reply .github/workflows/continuous-integration.yml:12]: Fun fact: github's yaml parser interprets any round floats as ints

File: lib/ogr/layer/extensions.rb (hunk 2 above)
Comment (tindron) [reply lib/ogr/layer/extensions.rb:116]: Yeah, the actual fix that's going to be used is in https://github.com/agrian-inc/ffi-gdal/pull/42/files, but on develop it's been moved over to this gem

**Review by roaming-rockenfels — APPROVED** [review]

**Review by turboladen — APPROVED** [review]

**Pull request merged**